### PR TITLE
Migrate to new UIKit design tokens

### PR DESCRIPTION
**Pull request by emmenko**
In UIKit `v15.3.0` we refactored the internal usage of design tokens to provide better support for theming.
This was the first step to enable the work in the design system and MC redesign.

In short, we rely on CSS variables to define the design tokens and to override them (when defining a new theme).

In theory all design tokens are backwards compatible but in the DocsKit we are using our own abstraction on top of it, so we need to adjust certain things.
For instance, we don't need to use Emotion theming anymore and instead rely on using and overwriting CSS variables. This makes the setup simpler.

The changes here are mostly about replacing Emotion theming with CSS variable overrides.

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
-    "target": "es2019",
     "jsxImportSource": "@emotion/react",
+    "lib": ["ESNext", "DOM"],
     "module": "ESNext",
     "moduleResolution": "Node",
-    "esModuleInterop": true,
     "strict": true,
     "skipLibCheck": true,
-    "allowJs": true
+    "target": "ES2022"
```
Comment (gabriele-ct) [inline tsconfig.json:12]: shall we concern about browser compatibility here?

**Review by gabriele-ct — APPROVED** [review]
to the best of my knowledge all seems fine 👍

**Review by gabriele-ct — APPROVED** [review]

**Comment by nkuehn** [comment]
Can one of you that has reviewed this in detail add the changeset file with the respective compatibility?  I can only guess whether this is minor / major / patch. Thanks !

**Comment by emmenko** [comment]
Bumping `patch` should be enough

**Pull request merged**